main.tf

```diff
@@ -249,8 +249,8 @@ data "aws_iam_policy_document" "default" {
     actions = distinct(compact(var.iam_actions))
 
     resources = [
-      join("", aws_elasticsearch_domain.default.*.arn),
-      "${join("", aws_elasticsearch_domain.default.*.arn)}/*"
+      join("", aws_elasticsearch_domain.default[*].arn),
+      "${join("", aws_elasticsearch_domain.default[*].arn)}/*"
```

Comment:
Note that here is a case where we do NOT want to use `one()`, because a resulting `null` is an invalid value for the string template.

Reply:
correct